src/jit/hwintrinsicxarch.cpp

```diff
@@ -234,6 +234,54 @@ int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: get the number of arguments based on HW GenTree node
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function
+//    node      -- GenTreeHWIntrinsic* node
+//
+// Return Value:
+//     number of arguments
+//
+int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, GenTreeHWIntrinsic* node)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+
+    int numArgs = hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
+    if (numArgs != -1)
+    {
+        return numArgs;
+    }
+
+    assert(numArgs == -1);
+
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+    if (op1 != nullptr)
+    {
+        if (op2 != nullptr)
```

Comment:
nit: You can move this check above the `if(op1 != nullptr)` check and it will be slightly more efficient:
```
if (op2 != nullptr)
{
    return 2;
}

if (op1 != nullptr)
{
    if(op1->OperIsList())
    {
    }
    else
    {
        return 1;
    }
}
```